ui-core/src/components/inputs/InputWithSuggestions.tsx

```diff
@@ -0,0 +1,107 @@
+import React, {
+  useState,
+  useRef,
+  useEffect,
+  ChangeEventHandler,
+  KeyboardEventHandler,
+  ReactNode,
+} from 'react';
+import cx from 'classnames';
+import Input, { InputProps } from './Input';
+
+interface InputWithSuggestionsProps<T> extends Omit<InputProps, 'onChange' | 'value'> {
+  suggestions: Array<T>;
+  onChange: (value: T) => void;
+  getSuggestionLabel: (option: T) => string;
+  getSuggestionKey: (value: T) => string;
```

Comment:
Not sure to understand how these props work ? Will we need to pass them when using this input in the app ?

Reply:
it's for consistency purposes as we use the same pattern in the Select component in order to pass a generic to the listing components

Reply:
Probably it's related to what you asked @kmer2016. 

But I'm not sure to see what is the difference between having the `onChange` which updates both `label` and `value` at once and also having the `getLabel` and `getValue` ?

Reply:
The onChange function does not update the label and value. Its purpose is to send the selected suggestion back to the parent component. The getSuggestionLabel and getSuggestionKey functions allow the component to manage the display of each suggestion, which can be of any type.